pc_helper_online.py

```diff
@@ -47,6 +51,23 @@ def main():
                     apply_publiccloud_pint_image(settings)
                     if "PUBLIC_CLOUD_IMAGE_ID" not in settings:
                         log.error(f"Failed to get PUBLIC_CLOUD_IMAGE_ID from {data}")
+                if "SLES4SAP_PINT_QUERY" in settings and "incidents" in data:
+                    for flavor in data["incidents"]["FLAVOR"]:
+                        if (
+                            "pint_name_regexp"
+                            not in data["incidents"]["FLAVOR"][flavor]
+                        ):
+                            continue
+                        cloud_provider = ""
+                        if "azure" in flavor.lower():
+                            cloud_provider = "AZURE"
```

Comment:
Why not
```
cloud_provider = flavor.lower()
```
Instead of these 3 lines? Do we need `cloud_provider` to be empty on GCE and EC2?

Reply:
To be more flexible

TL;DR
I'm not even sure that inspecting FLAVOR is a good idea at all. Problem is that we do not have any strongly enforced policy in the sles4sap metadata files about the flavor names for cloud build. Usually we fit in there the cloud provider name, we fit the BYOS/PAYG. So I decided to use this more generic test criteria to be more flexible. Another possible approach is not to use the FLAVOR at all and add an explicit.